src/script/scripting_server.cpp

```diff
@@ -88,6 +89,48 @@ ServerScripting::ServerScripting(Server* server):
 	infostream << "SCRIPTAPI: Initialized game modules" << std::endl;
 }
 
+void ServerScripting::initAsync()
+{
+	// Save globals to transfer
+	{
+		lua_State *L = getStack();
+		lua_getglobal(L, "core");
+		luaL_checktype(L, -1, LUA_TTABLE);
+		lua_getfield(L, -1, "get_globals_to_transfer");
+		lua_call(L, 0, 1);
+		luaL_checktype(L, -1, LUA_TSTRING);
+		getServer()->m_async_globals_data.set(readParam<std::string>(L, -1));
+		lua_pushnil(L);
+		lua_setfield(L, -3, "get_globals_to_transfer"); // unset function too
+		lua_pop(L, 2); // pop 'core', return value
+	}
+	
+
+	infostream << "SCRIPTAPI: Initializing async engine" << std::endl;
+	asyncEngine.registerStateInitializer(InitializeAsync);
+	asyncEngine.registerStateInitializer(ModApiUtil::InitializeAsync);
+	asyncEngine.registerStateInitializer(ModApiCraft::InitializeAsync);
+	asyncEngine.registerStateInitializer(ModApiItemMod::InitializeAsync);
+	asyncEngine.registerStateInitializer(ModApiServer::InitializeAsync);
+	// not added: ModApiMapgen is a minefield for thread safety
+	// not added: ModApiHttp has no way to match the callbacks
```

Comment:
Could you only allow the handle-based HTTP API, and not the callback-based HTTP API? I suppose that HTTP is already callback based/async so there's not a huge amount of benefit to having it, can be left for the future

Reply:
I rechecked and the actual reason is that there is no `core.after`, the handle-based API would work of course but is pointless without being able to delay for some time. Async jobs are also restricted to immediately returning a value...

Reply:
Makes sense